### ISSUE TITLE
Terminal features wishlist

### ISSUE THREAD

**Issue by siduck76** [NvChad/NvChad#247]
![image](https://user-images.githubusercontent.com/59060246/129441814-4dc0c41a-bf9c-4c18-81bc-0f6066faf1f1.png)

since toggle term just toggles and untoggles single terminal buffers , we need mappings to manually spawn multiple terminals .

**Comment by G-Rowell** [comment]
Assign me!

**Comment by G-Rowell** [comment]
@siduck76 

Time to close this?

**Comment by siduck76** [comment]
- open multiple terminals in different directions. Since we can just open one terminal with toggleterm , like open a terminal vertically , try opening another terminal horizontally , the previously vertical terminal will get toggled. We dont want that . The old term setup had the functionality of spawning multiple terminals . Toggleterm currently unsupoorts this i.e opening many terminals in different directions , it can only keep one terminal opened.

**Comment by G-Rowell** [comment]
So you haven't really listed the requirements/what you want it to do,
maybe consider something like:

# Wishlist
- mapping to create a new terminal in vert & horizontal directions
- mapping to create a buffer terminal
- mapping to do X
- able to hide a specific terminal


What I've asked is for you to describe in dot points what you want, there is literally only one dot point & a huge paragraph there

**Comment by G-Rowell** [comment]
I suggest closing https://github.com/siduck76/NvChad/issues/249 & https://github.com/siduck76/NvChad/issues/248

And rename this as `Terminal features/tools wishlist`

**Comment by siduck76** [comment]
## Todo

- mapping to create terminal in vert , horizontal directions  ( split )
- mapping to create terminal buffer ( should show the buffer in bufferline)
- Being able to toggle a specific terminal ( doesnt matter if its nolistedbuf or a buffer terminal)
- Being able to untoggle terminal buffers or nolistedbuf terminals from telescope by selecting them.

**Comment by G-Rowell** [comment]
>Being able to toggle a specific terminal ( doesnt matter if its nolistedbuf or a buffer terminal)
Being able to untoggle terminal buffers or nolistedbuf terminals from telescope by selecting them.
1. Can you clarify what you mean by `toggle` and `untoggle` in this context?

> nolistedbuf or a buffer terminal
2. Can you clarify what you mean by this? (or what your conceptual idea here is)

**Comment by siduck76** [comment]
> > Being able to toggle a specific terminal ( doesnt matter if its nolistedbuf or a buffer terminal)
> > Being able to untoggle terminal buffers or nolistedbuf terminals from telescope by selecting them.
> 
> 1. Can you clarify what you mean by `toggle` and `untoggle` in this context?
> 
> > nolistedbuf or a buffer terminal
> 
> 1. Can you clarify what you mean by this? (or what your conceptual idea here is)

- toggle = hide the terminal completely 
- untoggle = unhide it ( dont worry about the direction ) 
- we could use telescope to untoggle and let the user set direction for untoggling it ( ctrl + v or ctrl +x )

- for example when you try :telescope buffers :
- pressing ctrl +v on the selected item wil open the item vertically


![image](https://user-images.githubusercontent.com/59060246/129470190-d41ea193-7146-4b04-b4dd-4bdcfd995954.png)

**Comment by G-Rowell** [comment]
@siduck76 

Ah awesome! 
Your comment really clarifies what you're thinking, thank you, it's great.

May ask more soon but ty :heart:

**Comment by siduck76** [comment]
> @siduck76
> 
> Ah awesome!
> Your comment really clarifies what you're thinking, thank you, it's great.
> 
> May ask more soon but ty

Its day here so there's enough light in my room and actually read your comments properly and be sane :rofl:

**Comment by siduck76** [comment]
## NEW WISHLIST 

- add previous ( old ) mappings to open terminals in two directions ( hori and vert).
- hide a terminal  wheter its a buffer or nolistedbuffer ( if its a terminal window then its buffer shouldnt be visible in bufferline )
- open the hidden nolisted or terminal buffer window with telescope.

**Comment by G-Rowell** [comment]
So currently it's looking like on the larger problems is remembering where to put a terminal buffer that we want to `unhide`.
I'm trying to brainstorm a reasonable way to choose where it goes back to?

Like we can hide a buffer terminal, sure. 
But there is no nice way to put them back unless we wrap them in our own Lua object, exactly like `nvim-toggleterm.lua` does. 

:/ 

Note, I currently do have commands to `hide` and `un-hide` terminal buffers.

Any opinions @Akianonymus?

**Comment by siduck76** [comment]
> So currently it's looking like on the larger problems is remembering where to put a terminal buffer that we want to `unhide`.
> I'm trying to brainstorm a reasonable way to choose where it goes back to?
> 
> Like we can hide a buffer terminal, sure.
> But there is no nice way to put them back unless we wrap them in our own Lua object, exactly like `nvim-toggleterm.lua` does.
> 
> :/
> 
> Note, I currently do have commands to `hide` and `un-hide` terminal buffers.
> 
> Any opinions @Akianonymus?

Idk much about it , maybe you can store them in :ls! , also if there's a way to open no listed buffers with telescope then its gonna be easy to un-hide them.

**Comment by G-Rowell** [comment]
Ahhhh, 

my point is


There is no way to tell if a `hidden terminal` came from a split, vertical split or separate window/buffer

So there is no great way to tell them apart when restoring them

**Comment by siduck76** [comment]
> Ahhhh,
> 
> my point is
> 
> There is no way to tell if a `hidden terminal` came from a split, vertical split or separate window/buffer
> 
> So there is no great way to tell them apart when restoring them

Dont worry , as I told we will be restoring them via telescope so it depends on the user that how he restores them.

-  Suppose there are list of hidden terminal buffers and the user views them with telescope , just like :telescope buffer .
-  When a hidden terminal buffer is selected in the list and the user presses enter , the terminal buffer opens as window.
- If the user presses  Ctrl - v on the selected terminal buffer ( in telescope bruh ) then it restores vertically .
-  If the user presses Ctrl - x , then it restores horizontally

SO BASICALLY TELESCOPE WILL HANDLE ITSELF HOW TO OPEN THE TERMINAL BUFFER.

**Comment by siduck76** [comment]
oops pressed the close issue by mistake xD

**Comment by siduck76** [comment]
I told you already , we will keep both toggleterm and previous terminal setup . 

- toggle term is limited ( can handle 3 terminals only ) so its gonna be used for automatic term toggles since it also manages the directions.

- Chad old terminal setup , this is for manual hiding / unhiding of terminals .

**Comment by G-Rowell** [comment]
Maybe....just maybe can make a poorly coded, but better working terminal system 

BIG HMMMMMMMMMMMMMMMMMMMMMMM ENERGY

**Comment by Akianonymus** [comment]
> Any opinions @Akianonymus?

No idea 😂

**Comment by G-Rowell** [comment]
Closing, implemented by PR: https://github.com/NvChad/NvChad/pull/275

Discussion can continue there
